### ISSUE TITLE
Share To using Apple Shortcuts

### ISSUE THREAD

**Issue by bmpreston** [Wingysam/Christmas-Community#132]
Is there anyway to essentially create an apple shortcuts app, that I can put into the Share To to allow me to 'share' a new item into the webapp?

**Comment by Wingysam** [comment]
It could technically be done I think! It could break if Christmas Community changes how adding things works in the future though. I'll give it a go and let you know what I come up with.

**Comment by Wingysam** [comment]
I got it working: https://www.icloud.com/shortcuts/f437f6331c794ce9a3ceeb68d7b95dfe

Keep in mind that if the way the page works changes in the future this will stop working. It also requires you to extract the Cookie header from the site.

**Comment by bmpreston** [comment]
This is amazing, except can you help point me to the cookie header?  Is this the same from device to device, based on the logged in user?  ie if I find it on my mac, and transfer this to my iphone, would it still work?  

Thank you for you effort!

**Comment by Wingysam** [comment]
You're welcome!

The cookie from any device will work as long as it's signed into your account. You can use the network device tools on your Mac to get the cookie, or you can use this JavaScript in the console on your Mac:
```js
console.log(document.cookie)
```

**Comment by bmpreston** [comment]
Am I looking for the christmas_community.connect.sid?

I am stealing all of the code that the console.log dumped, and putting it into chatgpt.  I am horribly mentally underpowered to handle this information.  😂

--I might have Asperger's but I ain't no damned genius...  🤪

**Comment by Wingysam** [comment]
Yes, that's the cookie you're looking for. You can technically just dump the entire blob from the console.log, but if you have other stuff running on the same IP/port it'll have excess data. The answer to the prompt you get for the cookie should look like this:
```
christmas_community.connect.sid=<some random data>
```

**Comment by bmpreston** [comment]
That SID should remain during browser restarts, server and docker restarts, etc?  I might only find it not working if you alter your code?  

If I set this as a text value in the Shortcut, it should forever remain until the end of time, and the dinosaurs reclaim this planet?

If you think that's true, you are awesome.  I need a link to your support/send a tip page.

Thank you!

**Comment by Wingysam** [comment]
I believe as long as you use it at least once every week (or whatever you set `SESSION_MAX_AGE` to) and don't use the "Log out" button on the browser you harvested it from, it will never expire.

If I'm mistaken, I can make a system for long-lived auth cookies that don't expire.

**Comment by bmpreston** [comment]
That would be ballin'.

**Comment by Wingysam** [comment]
Let me know if it's still working in a week please! If you'd like to donate, the link is at the bottom of the README.

**Comment by bmpreston** [comment]
Doesn't seem to work.  I added the url, added what I believe is the cookie per the SID grab, and nothing when I visit and 'share from Amazon'.  Lemme know if you have other suggestions...

**Comment by jskiddie** [comment]
@bmpreston wingysams shortcut works for for me :)
cookie comes in the form of: `christmas_community.connect.sid=tfzdquguzutBu2t37uzg234Kdgfksdfsd` and the url in the form of `https://wishes.example.com/wishlist/john`
I have created an improved/easier to use version, provided you are able to install (and if necessary link) the two other shortcuts it depends on as well as scriptable: https://itunes.apple.com/us/app/scriptable/id1405459188
Main Shortcut: "`Christmas Community easy.shortcut`"(`09F092553C60FE5CD0E3DE49FB21F740`) 
Two dependent shortcuts: "`M.D. mkdir -p.shortcut`"(`4762670BF9C7BF0D5ECB8906D90EDDDA`), "`M.D. check local path exists bool opti.shortcut`"(`B2AC4C290D9DE70FD1D097AD49536F47`)
all md5
[ThreeShortcuts.zip](https://github.com/Wingysam/Christmas-Community/files/15340201/ThreeShortcuts.zip)

**Comment by jskiddie** [comment]
@bmpreston wingysams shortcut works for for me :)
cookie comes in the form of: christmas_community.connect.sid=tfzdquguzutBu2t37uzg234Kdgfksdfsd and the url in the form of https://wishes.example.com/wishlist/john
I have created an improved/easier to use version, provided you are able to install (and if necessary link) the two other shortcuts it depends on as well as scriptable: https://itunes.apple.com/us/app/scriptable/id1405459188
Main Shortcut: "Christmas Community v2.2.shortcut"(1938F4C6F26607CC856FFFD20CB2BDDA)
Two dependent shortcuts: "M.D. mkdir -p.shortcut"(4762670BF9C7BF0D5ECB8906D90EDDDA), "M.D. check local path exists bool opti.shortcut"(B2AC4C290D9DE70FD1D097AD49536F47)
all md5
[ThreeShortcutsv0.0.3.zip](https://github.com/Wingysam/Christmas-Community/files/15340319/ThreeShortcutsv0.0.3.zip)
